app/controllers/application_controller.rb

```diff
@@ -15,10 +15,21 @@ class ApplicationController < ActionController::Base
   rescue_from ActiveRecord::RecordNotFound, with: :not_found!
 
   def current_organization
-    @current_organization ||= Organization.find_by(short_name: params[:organization_id]) || current_user&.organization
+    return @current_organization if @current_organization
+
+    return current_role.resource if current_role.resource.is_a?(Organization)
+
+    Organization.find_by(short_name: params[:organization_id])
```

Comment:
I'm not sure why you aren't setting @current_organization if it isn't already set, in the new version.    That feels like it could have some side effects.

Reply:
Nope. Did a search and nothing else reads this instance variable (as it should be).